File: java/src/test/java/testingil/webinar/tddapi/s09/refactor_first_test/NavigatorUnitTests.java
package testingil.webinar.tddapi.s09.refactor_first_test;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import testingil.webinar.tddapi.s09.Distance;
import testingil.webinar.tddapi.s09.DistanceProvider;
import testingil.webinar.tddapi.s09.Location;
import testingil.webinar.tddapi.s09.Navigator;

import static org.hamcrest.MatcherAssert.assertThat;


class NavigatorUnitTests {

	@Test
	void when_created_distance_is_always_zero() {
		Location loc = new Location("London");
		DistanceProvider distService = new DistanceProvider();
		Navigator nav = new Navigator(loc, distService);
		Distance distance = nav.getDistanceFromDestination();
		assertThat(distance.inKm(), is(0));
	}
	
//	@Test
//	@DisplayName("2500 miles = 4000 kms")
//	void distance_is_calculated_in_km() {
//		Location startingPoint = new Location("New York City");
//		Navigator nav = new Navigator(startingPoint);
//		Location dest = new Location("Los Angeles");
//		nav.setDestination(dest);
//		Distance distance = nav.getDistanceFromDestination();
//		assertThat(distance.inKm(), is(4000));
//	}

}

File: Python/s09_refactor_first_test/navigator_unit_tests.py
import unittest

from hamcrest import assert_that, equal_to

from s09_refactor_first_test.distance_adapter import DistanceAdapter
from s09_refactor_first_test.location import Location
from s09_refactor_first_test.navigator import Navigator


class NavigatorUnitTests(unittest.TestCase):
    def test_when_created_distance_is_always_zero(self):
        loc = Location("London")
        distProvider = DistanceAdapter()
        nav = Navigator(loc, distProvider)
        distance = nav.get_distance_from_destination()
        assert_that(distance.inKm(), equal_to(0))

    # def test_distance_is_calculated_in_km(self):
    #     """2500 miles = 4000 kms"""
    #     startingPoint = Location("New York City")
    #     nav = Navigator(startingPoint)
    #     dest = Location("Los Angeles")
    #     nav.setDestination(dest)
    #     distance = nav.getDistanceFromDestination()
    #     assert_that(distance.inKm(), equal_to(4000))


if __name__ == '__main__':
    unittest.main()
